server/backends/blobstore/util/util.go

```diff
@@ -121,3 +123,54 @@ func RecordDeleteMetrics(typeLabel string, startTime time.Time, err error) {
 		metrics.BlobstoreTypeLabel: typeLabel,
 	}).Observe(float64(duration.Microseconds()))
 }
+
+type compressedWriter struct {
+	ctx        context.Context
+	gzipWriter *gzip.Writer
+	closer     io.Closer
+	label      string
+	n          int
+}
+
+func NewCompressedBlobStoreWriter(ctx context.Context, w io.Writer, c io.Closer, label string) io.WriteCloser {
+	return &compressedWriter{ctx: ctx, gzipWriter: gzip.NewWriter(w), closer: c, label: label, n: 0}
+}
+
+func (d *compressedWriter) Write(p []byte) (int, error) {
+	start := time.Now()
+	_, spn := tracing.StartSpan(d.ctx)
+	n, err := d.gzipWriter.Write(p)
+	spn.End()
+	RecordWriteMetrics(d.label, start, d.n, err)
```

Comment:
we might want to omit these Write metrics for now -- the metrics could get skewed in the case where the gzipWriter is doing internal buffering/async flushing (and the metrics would show that the Write() completes in nanoseconds, even the bytes have just been moved from one in-memory buffer to another).

Another option is to do it in `Close()` but that is also not ideal because it depends on client upload latency+throughput (while the existing usages of this metric only depend on the server -> blobstore latency+throughput).